pennylane/interfaces/torch.py

```diff
@@ -246,3 +258,289 @@ def execute(tapes, device, execute_fn, gradient_fn, gradient_kwargs, _n=1, max_d
+
+def _compute_vjps_new(dys, jacs, multi_measurements, device=None):
+    """Compute the vjps of multiple tapes, directly for a Jacobian and tangents."""
+    # pylint: disable=unused-argument
+    # TODO: add device
+    vjps = []
+
+    for i, multi in enumerate(multi_measurements):
+        compute_func = (
+            qml.gradients.compute_vjp_multi_new if multi else qml.gradients.compute_vjp_single_new
+        )
+        vjps.extend(compute_func(dys[i], jacs[i]))
+    return vjps
+
+
+@pytreeify
+class ExecuteTapesNew(torch.autograd.Function):
+    """The signature of this ``torch.autograd.Function`` is designed to
+    work around Torch restrictions.
+
+    In particular, ``torch.autograd.Function``:
+
+    - Cannot accept keyword arguments. As a result, we pass a dictionary
+      as the first argument ``kwargs``. This dictionary **must** contain:
+
+      * ``"tapes"``: the quantum tapes to batch evaluate
+      * ``"device"``: the quantum device to use to evaluate the tapes
+      * ``"execute_fn"``: the execution function to use on forward passes
+      * ``"gradient_fn"``: the gradient transform function to use
+        for backward passes
+      * ``"gradient_kwargs"``: gradient keyword arguments to pass to the
+        gradient function
+      * ``"max_diff``: the maximum order of derivatives to support
+
+    Further, note that the ``parameters`` argument is dependent on the
+    ``tapes``; this function should always be called
+    with the parameters extracted directly from the tapes as follows:
+
+    >>> parameters = []
+    >>> [parameters.extend(t.get_parameters()) for t in tapes]
+    >>> kwargs = {"tapes": tapes, "device": device, "gradient_fn": gradient_fn, ...}
+    >>> ExecuteTapes.apply(kwargs, *parameters)
+
+    The private argument ``_n`` is used to track nesting of derivatives, for example
+    if the nth-order derivative is requested. Do not set this argument unless you
+    understand the consequences!
+    """
+
+    @staticmethod
+    def forward(ctx, kwargs, *parameters):  # pylint: disable=arguments-differ
+        """Implements the forward pass batch tape evaluation."""
+        ctx.tapes = kwargs["tapes"]
+        ctx.device = kwargs["device"]
+
+        ctx.execute_fn = kwargs["execute_fn"]
+        ctx.gradient_fn = kwargs["gradient_fn"]
+
+        ctx.gradient_kwargs = kwargs["gradient_kwargs"]
+        ctx.max_diff = kwargs["max_diff"]
+        ctx._n = kwargs.get("_n", 1)
+
+        with qml.tape.Unwrap(*ctx.tapes):
+            res, ctx.jacs = ctx.execute_fn(ctx.tapes, **ctx.gradient_kwargs)
+
+        # if any input tensor uses the GPU, the output should as well
+        ctx.torch_device = None
+
+        for p in parameters:
+            if isinstance(p, torch.Tensor) and p.is_cuda:  # pragma: no cover
+                ctx.torch_device = p.get_device()
+                break
+
+        for i, r in enumerate(res):
+
+            res[i] = _res_to_torch(r, i, ctx)
+
+            _jac_to_torch(i, ctx)

```

Comment:
Can we add a comment here mentioning that `_jac_to_torch` will modify the contents of `ctx` in-place?

Reply:
Actually, can we use [ctx.save_for_backward()](https://pytorch.org/docs/stable/generated/torch.autograd.function.FunctionCtx.save_for_backward.html) here? Seems like it's the intended way to save values for the backward pass.

Reply:
I am adding a not for the in-place change! Unfortunately it only works for tensors.